packages/types/config/runtime.d.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * NuxtConfigurationRuntimeConfig
+ * Documentation : TBD
+ * NuxtRuntimeConfig interface can be extended by users to enable intellisense on $config
+ */
+
+export interface NuxtRuntimeConfig {
+  [key: string]: any
```

Comment:
Don't know about cons/pros but what about `[key: string]: NuxtRuntimeConfig | any`?

Reply:
@pi0 Why you would want to do that ? Do you have an example ?

Reply:
I think that would be wrong as then when someone extends `NuxtRuntimeConfig` like:
```
interface NuxtRuntimeConfig {
  something: ?bool
}
```
that `something` would effectively be repeated for all nested objects also.

Reply:
That said, I think it would be possible to improve that type still by restricting it somewhat and making clear that recursively nested objects are supported.

Reply:
My intention was to provide precise supported types. Other than all key levels should be string, end values should be POJO to allow serializing (no function or symbol, etc)

Reply:
Maybe something like:

```
type JsonVal = string | number | boolean | JsonObj | JsonObj[] | null;

export interface NuxtRuntimeConfig {
  [key: string]: JsonVal | JsonVal[];
}
```

Reply:
Seems fine :) We can also use something more compact like this: https://doc.deno.land/https/deno.land/x/destr/src/index.ts#JSONValue

Reply:
No because we want the `NuxtRuntimeConfig` to be extendable so it has to be an interface.

Reply:
Care to explain the revert?